core-bundle/contao/classes/Theme.php

```diff
@@ -715,8 +715,7 @@ protected function extractThemeFiles($arrFiles, $arrDbFields)
 		$objSession = System::getContainer()->get('request_stack')->getSession();
 		$objSession->remove('uploaded_themes');
 
-		$this->import(Automator::class, 'Automator');
-		$this->Automator->generateSymlinks();
+		System::importStatic(Automator::class)->generateSymlinks();
```

Comment:
@contao/developers Is this the way to go or would you prefer `(new Automator())->generateSymlinks()`?

Reply:
I would think either `Framework::getAdapter()` or `Framework::createInstance()` if we can't use DI ?

Reply:
also makes we wonder if we should use adapters instead of `System::importStatic` 😆

Reply:
I’d prefer `(new Automator())->generateSymlinks()`